src/obstacle_detector.cpp

```diff
@@ -217,6 +289,25 @@ void ObstacleDetector::get_points()
     obstacle_mutex.unlock();
 }
 
+// Get points from lidar - convert from polar to cartesian coords
+void ObstacleDetector::get_lidar_points(std::vector<tf2::Vector3>& points)
+{
+    obstacle_mutex.lock();
+    for (const auto& p : lidar_points) {
+        float sin_theta = std::sin(p.theta);
+        float cos_theta = std::cos(p.theta);
+
+        float x = lidar_origin.x() + p.radius * (lidar_normal.x() * cos_theta -
+             lidar_normal.y() * sin_theta);
+
+        float y = lidar_origin.y() + p.radius * (lidar_normal.y() * cos_theta +
+             lidar_normal.x() * sin_theta);
+
+        points.push_back(tf2::Vector3(x, y, 0));
```

Comment:
I think all of the code to compute this vector should be within PolarLine, in a function that returns a Vector3. (Which would mean that PolarLine would need the lidar origin and normal).

At least going from polar to Cartesian should be, and those points could be transformed to the robot coordinate system here (still a function that returns a vector3).

Reply:
The points are transformed to base_link here, that's what lidar_origin and lidar_normal do.

Reply:
I'm not sure if it makes much sense to have a class that has a point in `r, theta` form and has its origin in point normal form.  I think it's cleaner if the coordinate system transform is done outside the class.

Reply:
Ok, that makes sense.